### PR TITLE
feature(locksmith): Refactor test suite for scoped mocks and parallel execution

### PR DESCRIPTION
**Pull request by 0xTxbi**
# Description
This PR introduces improvements to the test suite by scoping previously global fetch mocks to individual tests, enhancing test isolation and reliability. Additionally, it optimizes the test configuration to enable parallel execution in isolated threads, ensuring each test operates within an independent environment for improved performance and consistency.

# Issues
Fixes #
Refs #

# Checklist:

- [x] 1 PR, 1 purpose: my Pull Request applies to a single purpose
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have updated the docs to reflect my changes if applicable
- [ ] I have added tests (and stories for frontend components) that prove my fix is effective or that my feature works
- [x] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

File: locksmith/vite.config.ts [hunk 1]
```diff
@@ -11,8 +11,16 @@ export default defineConfig({
...
     testTimeout: 1000 * 60 * 5,
+    // run tests in parallel in isolated threads
+    pool: 'threads',
+    poolOptions: {
+      threads: {
+        singleThread: false,
+      },
+    },
+    // isolate test environment for each test
+    isolate: true,
```
Comment (julien51) [inline locksmith/vite.config.ts:24]: I think we have that config in the package.json script as well!

Comment (julien51) [reply locksmith/vite.config.ts:24]: please check and clean up there?

Comment (julien51) [inline locksmith/vite.config.ts:24]: ok so this is the default anyway, so we can remove

**Comment by 0xTxbi** [comment]
closing this PR as the issue has been successfully addressed in #14992.

**Pull request closed without merging**